.github/workflows/ci.yaml

```
@@ -244,6 +244,15 @@ jobs:
           distribution: adopt
           java-version: 8
 
+      - name: Read .nvmrc
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: "${{ steps.nvm.outputs.NVMRC }}"
```

Did you know that `setup-node` comes with caching now?

```suggestion
          node-version: "${{ steps.nvm.outputs.NVMRC }}"
          cache: npm
```

Read the readme and learn more it here: https://github.com/actions/setup-node#caching-packages-dependencies.

By the way `setup-java` comes with caching as well, but I don't know much about Java, so won't apply any suggestions here.

https://github.com/actions/setup-java#caching-packages-dependencies

Also check if the version of `setup-java` you're using comes with the caching capability before you turn it on for Java.

---

That's a nice new feature. Thanks for letting us know. We'll have a look and see what we might be able to refactor, because the build definition sure is long.

---

@edalex-ian Do you want me to make an issue with the same text? That way I'm not blocking this PR, and you can track the issue that way, and follow up when you have time to work on it.

---

That'd be great thanks @HonkingGoose. :+1: 

I'll go and merge this in the meantime. :wink:

---

See issue #3390 for the "feature request".